### PR TITLE
PLATUI-803: Specify GTM container using data attribute

### PR DESCRIPTION
**Pull request by matthewmascord**

### CODE REVIEW

File: app/uk/gov/hmrc/trackingconsentfrontend/assets/src/common/pageHandler.ts [hunk 1]
```diff
@@ -1,10 +1,11 @@
 import enableGtm from '../interfaces/enableGtm'
 import { UserPreferences } from '../../types/UserPreferences'
 import isFeatureEnabled from "../interfaces/isFeatureEnabled";
-import {enableTrackingConsent} from "../constants/featureNames";
+import { enableTrackingConsent } from "../constants/featureNames";
+import getGtmContainerId from "./getGtmContainerId";
 
 const pageHandler = (document: HTMLDocument, userPreference: UserPreferences, pageRenderer, containerId) => {
-    enableGtm(containerId)
+    enableGtm(getGtmContainerId() || containerId)
```
Comment (matthewmascord) [inline app/uk/gov/hmrc/trackingconsentfrontend/assets/src/common/pageHandler.ts:8]: This logic means the existing links to tracking.js and tracking/transitional.js will continue to work for now without a data attribute. At some point we should throw an error for teams not being explicit about the container as there is no default.

File: conf/app.routes [hunk 2]
```diff
@@ -3,13 +3,6 @@
...
 ->         /hmrc-frontend            hmrcfrontend.Routes
 GET        /tracking.js              controllers.Assets.at(path="/public", file="a.js")
-GET        /tracking/a.js            controllers.Assets.at(path="/public", file="a.js")
-GET        /tracking/b.js            controllers.Assets.at(path="/public", file="b.js")
-GET        /tracking/c.js            controllers.Assets.at(path="/public", file="c.js")
-GET        /tracking/d.js            controllers.Assets.at(path="/public", file="d.js")
-GET        /tracking/e.js            controllers.Assets.at(path="/public", file="e.js")
-GET        /tracking/f.js            controllers.Assets.at(path="/public", file="f.js")
-GET        /tracking/sdes.js         controllers.Assets.at(path="/public", file="sdes.js")
 GET        /tracking/transitional.js controllers.Assets.at(path="/public", file="transitional.js")
```
Comment (matthewmascord) [inline conf/app.routes:6]: Keeping this for now so as not to break existing users.

**Review by JoPintoPaul — APPROVED** [review]

File: app/uk/gov/hmrc/trackingconsentfrontend/assets/src/common/pageHandler.ts (hunk 1 above)
Comment (matthewmascord) [reply app/uk/gov/hmrc/trackingconsentfrontend/assets/src/common/pageHandler.ts:9]: @nat.carey suggested adding a deprecation notice in the console

File: app/uk/gov/hmrc/trackingconsentfrontend/assets/src/interfaces/enableGtm.ts [hunk 3]
```diff
@@ -3,7 +3,10 @@ import getNonce from "../common/getNonce";
 const dataLayer = 'dataLayer'
 const gtmBaseUrl = 'https://www.googletagmanager.com/gtm.js?id='
 
-const enableGtm = (containerId: string) => {
+const enableGtm = (containerId: string | undefined) => {
+  if (containerId === undefined) {
+    throw "Unable to enable GTM because no container has been specified"
```
Comment (matthewmascord) [inline app/uk/gov/hmrc/trackingconsentfrontend/assets/src/interfaces/enableGtm.ts:8]: add new Error()

**Review by natcarey — APPROVED** [review]

**Pull request merged**